File: CookerConnectorAndroid/app/src/main/java/com/schoovello/cookerconnector/data/FirebaseDbInstance.java
package com.schoovello.cookerconnector.data;

import com.google.firebase.database.FirebaseDatabase;

public final class FirebaseDbInstance {

	private static FirebaseDatabase sInstance;

	public static synchronized FirebaseDatabase get() {
		if (sInstance == null) {
			sInstance = FirebaseDatabase.getInstance();
			sInstance.setPersistenceEnabled(false);
		}
		return sInstance;
	}


	private FirebaseDbInstance() {
		// do not instantiate
	}

}

File: CookerConnectorAndroid/app/src/main/java/com/schoovello/cookerconnector/data/FirebaseDbInstance.kt
package com.schoovello.cookerconnector.data;

import com.google.firebase.database.FirebaseDatabase;

object FirebaseDbInstance {

	val instance: FirebaseDatabase by lazy {
		FirebaseDatabase.getInstance().apply {
			setPersistenceEnabled(true)
			setPersistenceCacheSizeBytes(1024 * 1024 * 64) // 64MB
		}
	}
}
